test/TempusController.test.ts

```diff
@@ -314,6 +315,93 @@ describeForEachPool("TempusController", (testPool:PoolTestFixture) =>
+        testPool, 
+        owner, 
+        "0", 
+        "0",
+        await testPool.yields.balanceOf(owner),
+        false,
+        currentYieldsRate.toString(),
+        "0.03"
+      )).to.emit(testPool.amm.vault, 'Swap');;
+    });
+    
+    it("Should successfully swap Principals --> Yields w/ 3% Maximum Slippage", async () => 
+    {
+      await initAMM(user1, /*ybtDeposit*/1000000, /*principals*/100000, /*yields*/1000000);
+      await testPool.controller.depositYieldBearing(owner, testPool.tempus, 100);
+
+      const pricePerYield = await testPool.yields.getPricePerFullShareStored();
+      const pricePerPrincipal = await testPool.principals.getPricePerFullShareStored();
+      
+      const currentYieldsRate = new Decimal(pricePerYield.toString()).div(pricePerPrincipal.toString()).toFixed(testPool.principals.decimals);
+      
+      expect(await controller.exitTempusAmmAndRedeem(
+        testPool, 
+        owner, 
+        "0", 
+        await testPool.principals.balanceOf(owner),
+        "0",
+        false,
+        currentYieldsRate.toString(),
+        "0.03"
+      )).to.emit(testPool.amm.vault, 'Swap');
+    });
+
+    it("Should fail swap due to minimum return Principals --> Yields w/ 0.1% Maximum Slippage", async () => 
+    {
+      await initAMM(user1, /*ybtDeposit*/1000000, /*principals*/100000, /*yields*/1000000);
+      await testPool.controller.depositYieldBearing(owner, testPool.tempus, 100);
+
+      const pricePerYield = await testPool.yields.getPricePerFullShareStored();
+      const pricePerPrincipal = await testPool.principals.getPricePerFullShareStored();
+      
+      const currentYieldsRate = new Decimal(pricePerYield.toString()).div(pricePerPrincipal.toString()).toFixed(testPool.principals.decimals); 
+
+      await expect(controller.exitTempusAmmAndRedeem(
+        testPool, 
+        owner, 
+        "0", 
+        await testPool.principals.balanceOf(owner),
+        "0",
+        false,
+        currentYieldsRate.toString(),
+        "0.001"
+      )).to.be.revertedWith("BAL#507");
+    });
+
+    it("Should fail swap due to minimum return Yields --> Principals w/ 0.1% Maximum Slippage", async () => 
+    {
+      await initAMM(user1, /*ybtDeposit*/1000000, /*principals*/100000, /*yields*/1000000);
+      await testPool.controller.depositYieldBearing(owner, testPool.tempus, 100);
+
+      const pricePerYield = await testPool.yields.getPricePerFullShareStored();
+      const pricePerPrincipal = await testPool.principals.getPricePerFullShareStored();
+      
+      const currentYieldsRate = new Decimal(pricePerYield.toString()).div(pricePerPrincipal.toString()).toFixed(testPool.principals.decimals);
+
+      await expect(controller.exitTempusAmmAndRedeem(
+        testPool, 
+        owner, 
+        "0", 
+        "0",
+        await testPool.yields.balanceOf(owner),
+        false,
+        currentYieldsRate.toString(),
+        "0.001"
+      )).to.be.revertedWith("BAL#507");
```

Comment:
```suggestion
      await expect(
        controller.exitTempusAmmAndRedeem(testPool, owner, 0, 0, await testPool.yields.balanceOf(owner), false, currentYieldsRate, 0.001)
      ).to.be.revertedWith("BAL#507");
```
If you reformat the code a bit, it could look much more readable in this way, at least in my opinion

Reply:
I think we shouldn't worry about it tests formatting at all right now... I started working on a PR that adds eslint you our codebase so we just dictate some style rules and auto-enforce and fix all the styling issues we have...